parley/src/main/java/nu/parley/android/util/AccessibilityUtil.java

```diff
@@ -0,0 +1,192 @@
+            case MessageViewHolderFactory.MESSAGE_TYPE_LOADER:
+                return context.getString(R.string.parley_accessibility_message_loading);
+            case MessageViewHolderFactory.MESSAGE_TYPE_AGENT_TYPING:
+                return context.getString(R.string.parley_accessibility_message_agent_typing);
+            case MessageViewHolderFactory.MESSAGE_TYPE_DATE:
+            case MessageViewHolderFactory.MESSAGE_TYPE_INFO:
+            case MessageViewHolderFactory.MESSAGE_TYPE_MESSAGE_AUTO:
+            default:
+                return null;
+        }
+    }
+
+    @Nullable
+    private static String getContentDescriptionBody(Context context, Message message) {
+        StringBuilder builder = new StringBuilder();
+        // Title
+        if (!TextUtils.isEmpty(message.getTitle())) {
+            builder.append(message.getTitle());
+            appendDotIfNeeded(builder);
+        }
+        // Message
+        if (!TextUtils.isEmpty(message.getMessage())) {
+            builder.append(message.getMessage());
+            appendDotIfNeeded(builder);
+        }
+        // Media
+        if (message.hasImageContent()) {
+            builder.append(context.getString(R.string.parley_accessibility_message_media_attached));
+        }
+        // Actions
+        if (message.hasActionsContent()) {
+            builder.append(context.getString(R.string.parley_accessibility_message_actions_attached));
+        }
+        // Carousel
+        if (message.hasCarouselContent()) {
+            Message first = message.getCarousel().get(0);
+            @Nullable String content = getContentDescriptionBody(context, first);
+            if (content != null) {
+                builder.append(content);
+            }
+        }
+        // Return
+        if (builder.length() == 0) {
+            return null;
+        } else {
+            return builder.toString();
+        }
+    }
+
+    @Nullable
+    private static String getContentDescriptionEnding(Context context, Message message) {
+        StringBuilder builder = new StringBuilder();
+        switch (message.getSendStatus()) {
+            case Message.SEND_STATUS_FAILED:
+                builder.append(context.getString(R.string.parley_accessibility_message_failed));
+            case Message.SEND_STATUS_PENDING:
+                builder.append(context.getString(R.string.parley_accessibility_message_pending));
+            case Message.SEND_STATUS_SUCCESS:
+            default:
+                break;
+        }
+        if (message.getDate() != null) {
+            builder.append(context.getString(R.string.parley_accessibility_message_time));
+            builder.append(" ");
+            builder.append(DateUtil.formatTime(message.getDate()));
+        }
+        if (builder.length() == 0) {
+            return null;
+        } else {
+            return builder.toString();
+        }
+    }
+
+    private static final List<String> endChars = Arrays.asList(".", "?", "!");
```

Comment:
Shouldn't vars and statics be at the top?
Maybe also consider `punctuationChars` as an alternavite name.